lib/matplotlib/cm.py

```diff
@@ -65,12 +65,16 @@ def _gen_cmap_d():
     # Generate reversed cmaps.
     for cmap in list(cmap_d.values()):
         rmap = cmap.reversed()
+        cmap._global = True
+        rmap._global = True
         cmap_d[rmap.name] = rmap
     return cmap_d
 
 
-cmap_d = _gen_cmap_d()
-locals().update(cmap_d)
+_cmap_d = _gen_cmap_d()
+locals().update(_cmap_d)
+# This is no longer considered public API
+cmap_d = _cmap_d
```

Comment:
I wasn't sure how to warn on read/write here. Do I need to create a new Dictionary subclass that overwrites the getters/setters with the warn_deprecated call?

Reply:
I guess something like https://github.com/matplotlib/matplotlib/pull/10735 (simpler if Py3.7+) would be needed to fully warn on access (and then you don't need to warn on getitem/setitem I guess)?

Reply:
That is a nice helper! I think that may be very helpful for deprecating plt.cm usage. I decided to go the quick route and just subclass dict and add in warning on get/set only.